vendor/DOtherSide/lib/src/DOtherSide.cpp

```diff
@@ -371,6 +371,12 @@ void dos_qguiapplication_quit()
     QMetaObject::invokeMethod(qGuiApp, "quit", Qt::QueuedConnection);
 }
 
+void dos_qguiapplication_restart()
```

Comment:
I wonder if for things like that we need to go via nim layer. What about creating independent utility in cpp and expose it e.g. as a singleton to qml?

Reply:
I've had some thoughts about it. That was my first intention as well, but having the implementation in DOtherSide next to the `quit` implementation felt more intuitive now. As a result, the qml access needed to go needed to go through `utils.nim`. It's the place where I'd search for such an implementation (or anything else related to qGuiApp).

But we could add a task to move the entire utils.nim to cpp. WDYT? I think it's feasible.

Reply:
Yeah, that's good idea I think!